### ISSUE TITLE
ILogger usage confusing

### ISSUE THREAD

**Issue by Jaegermeiste** [danielwertheim/mynatsclient#84]
While implementing mynatsclient in a project over the past couple of days, I definitely spent a few minutes wondering why my ILogger wasn't passing to mynatsclient.

Either:
-  documentation should clarify that the ILogger isn't the standard .NET ILogger (Microsoft.Extensions.Logging) and subsequently will require a wrapper class
- or (better) mynatsclient should accept the standard Microsoft.Extensions.Logging.ILogger either along with or in lieu of the MyNatsClient.ILogger

**Comment by Jaegermeiste** [comment]
aside from documentation, fixed by #85

**Comment by Jaegermeiste** [comment]
If pull #85 is accepted, I propose the following update to the Logging section of documentation:

**Logging**
Some information is passed to a logger, e.g. Errors while trying to connect to a host. By default there's a MyNatsClient.NullLogger hooked in.

To hook a standard Microsoft.Extensions.Logging.ILogger, simply pass the logger to
`LoggerManager.UseMicrosoftLogger(myMicrosoftExtensionsLogger);` For more details on standard .NET Logging, see [https://docs.microsoft.com/en-us/dotnet/core/extensions/logging](https://docs.microsoft.com/en-us/dotnet/core/extensions/logging). Certain third party loggers such as Serilog can be hooked into the standard Microsoft.Extensions.Logging.ILogger, so this provides a simple means of extending that functionality to MyNatsClient.

To add a custom logger of choice, you must implement a class that derives from MyNatsClient.ILogger and assign a new resolver to LoggerManager.Resolve:
```
public class MyLogger : MyNatsClient.ILogger {
    public void void Trace(string message) { // Do logging }
    public void Debug(string message) { // Do logging}
    public void Info(string message) { // Do logging}
    public void Error(string message) { // Do logging}
    public void Error(string message, Exception ex) { // Do logging }
}

LoggerManager.Resolve = loggerForType => new MyLogger();
```
The loggerForType being passed could be used for passing to NLog to get Logger per class etc.
